### PR TITLE
[expo-cli] skip updating dependencies in package.json in `expo prebuild`

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

We need a way to preserve react-native version when building on EAS

# How

- add param to `expo prebuild`
- use comma separated list because variadic params are only supported in newer commander and updating commander version would break a lot of hacks we are using in cli

# Test Plan

`expo prebuild`
add tests

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]

**Comment by brentvatne** [comment]
could you add test coverage for this?

**Comment by brentvatne** [comment]
let's also wait for @EvanBacon to approve this before merging

File: packages/expo-cli/src/commands/eject/updatePackageJson.ts
Comment (EvanBacon) [inline packages/expo-cli/src/commands/eject/updatePackageJson.ts:27]:
```suggestion
  skipDependencyUpdate?: string[];
```

File: packages/expo-cli/src/commands/eject/updatePackageJson.ts
Comment (EvanBacon) [inline packages/expo-cli/src/commands/eject/updatePackageJson.ts:87]:
```suggestion
  skipDependencyUpdate?: string[];
```

File: packages/expo-cli/src/commands/prebuild.ts
Comment (EvanBacon) [inline packages/expo-cli/src/commands/prebuild.ts:53]:
```suggestion
      '--skip-dependency-update <dependencies>',
```

Comment (EvanBacon) [reply packages/expo-cli/src/commands/prebuild.ts:53]: i.e. it cannot be optional if it's defined.

File: packages/expo-cli/src/commands/prebuild.ts
Comment (EvanBacon) [inline packages/expo-cli/src/commands/prebuild.ts:54]: Do we use comma separated anywhere else? Does anyone have a preference towards collected values (`--skip-dependency-update react-native --skip-dependency-update expo-updates`)?

Comment (wkozyra95) [reply packages/expo-cli/src/commands/prebuild.ts:54]: > Do we use comma separated anywhere else?

No

> Does anyone have a preference towards collected values

It's the first thing I tried, but something like that is not supported in version of commander we are currently using and based on previous attempts updating commander version is not that easy.

File: packages/expo-cli/src/commands/eject/createNativeProjectsFromTemplateAsync.ts
Comment (wkozyra95) [reply packages/expo-cli/src/commands/eject/createNativeProjectsFromTemplateAsync.ts:47]: sure I can add that, but with the current code I'm passing `skipDependencyUpdate ?? []` at a certain point so the rest of the code does does not need to handle optional values

**Review by EvanBacon — APPROVED** [review]
Lgtm

**Pull request merged**
